edx/edx-enterprise
# ENT-265: Fix 500 server error appearing on enrolling user into course through manage learner page in Django Admin.

Hi @mattdrayer , @asadiqbal08 , @zubair-arbi 
Please take a look at this PR,

__Description:__
If we try to enroll a user to a course that does not have a start date (i.e. `None` value in course start date) using manage learner dashboard, we get 500 error. The issue is reproducible on stage only.
Here is the stack trace from Splunk,
```
Apr  5 10:29:34 ip-10-3-11-249 [service_variant=lms][django.request][env:stage-edx-edxapp] ERROR [ip-10-3-11-249  2705] [base.py:256] - Internal Server Error: /admin/enterprise/enterprisecustomer/b0990312-c5a3-48c4-b0fe-b8d844249623/manage_learners
Traceback (most recent call last):
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/utils/decorators.py", line 145, in inner
    return func(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/utils/decorators.py", line 110, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/views/decorators/cache.py", line 57, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/contrib/admin/sites.py", line 233, in inner
    return view(request, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/views/generic/base.py", line 71, in view
    return self.dispatch(request, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/newrelic-2.78.0.57/newrelic/hooks/framework_django.py", line 840, in wrapper
    return wrapped(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/views/generic/base.py", line 89, in dispatch
    return handler(request, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/enterprise/admin/views.py", line 765, in post
    notify=notify,
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/enterprise/admin/views.py", line 660, in _enroll_users
    users=all_successes,
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/enterprise/admin/views.py", line 502, in notify_enrolled_learners
    course_start = parse_lms_api_datetime(course_details.get('start'))
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/enterprise/lms_api.py", line 334, in parse_lms_api_datetime
    date_time = datetime.datetime.strptime(datetime_string, datetime_format)
TypeError: must be string, not None
```
**JIRA:** https://openedx.atlassian.net/browse/ENT-265

**Dependencies:**
1. Create a new course by visiting https://studio-{domain}/home/, you can use any existing course.
2. Go to CAT tool at https://ecommerce-{domain}/courses/new/ and add the course with some course type.
3. Visit course runs on discovery admin at https://discovery-{domain}/admin/course_metadata/courserun/ , open the course run corresponding the course you just added. Remove course start date and save.

**Note:** We do not have to go through above steps on stage as there are already course runs that do not have start date, these are the course runs causing the problem this PR is trying to fix.

**Testing instructions:**
1. Login to Django admin.
2. Navigate to Enterprise customer page.
3. Select an enterprise customer from list and click "Manage Learner" button.
4. Enter email in field and course ID (select the one you create above) in field and select mode of course .
5. Click submit button and view.
6. Learner should enroll into course on clicking submit button.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:** @mattdrayer we need to think about the case when course run does not have a start date, currently we are sending an email to user with the following test.
```
Hi!
You have been enrolled in Test Course, a course offered by Test Enterprise Customer. This course begins Dec. 31, 2029. For more information, see the following link:
http://open.edx:8000/courses/course-v1:edX+TC101+2017_T1/about
Thanks,
```
what should we (in the place of course start date) pass to email context when start date is None ?
